spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -82,6 +82,8 @@ public Long endSnapshotId() {
     return confParser.longConf().option(SparkReadOptions.END_SNAPSHOT_ID).parseOptional();
   }
 
+  /** @deprecated will be removed in 1.3.0; specify read branch in the table identifier */
+  @Deprecated
```

Comment:
Is this fine to be deprecated? We cannot read table by spark options at all in this case. My understanding was we use table identifiers for only write options and we leave read as it is.